oak_functions/lookup_data_generator/src/main.rs

```diff
@@ -52,11 +64,42 @@ fn main() -> anyhow::Result<()> {
     let opt = Opt::from_args();
     let mut buf = BytesMut::new();
     let mut rng = rand::thread_rng();
-    for _ in 0..opt.entries {
-        let entry = create_entry(&mut rng, opt.key_size_bytes, opt.value_size_bytes);
-        entry
-            .encode_length_delimited(&mut buf)
-            .context("could not encode entry")?;
+    match opt.cmd {
+        Command::Random {
+            key_size_bytes,
+            value_size_bytes,
+            entries,
+        } => {
+            for _ in 0..entries {
+                let entry = create_entry(&mut rng, key_size_bytes, value_size_bytes);
+                entry
+                    .encode_length_delimited(&mut buf)
+                    .context("could not encode entry")?;
+            }
+        }
+        Command::Weather {} => {
+            #[derive(Serialize)]
+            struct WeatherValue {
+                temperature_degrees_celsius: i32,
+            }
+            let dist = rand::distributions::Uniform::new(-30, 40);
+            for lat in -90..=90 {
+                for lon in -180..=180 {
+                    let key = format!("{},{}", lat, lon);
```

Comment:
nit: maybe move lines 88-96 to a `create_weather_entry` to be more consistent with the other arm of the match.

Reply:
Done.